Signal/src/ViewControllers/MessageMetadataViewController.swift

```diff
@@ -267,76 +276,79 @@ class MessageMetadataViewController: OWSViewController {
         }
     }
 
+    private func displayableTextIfText() -> String? {
+        let messageCellType = viewItem.messageCellType()
+        guard messageCellType == .textMessage ||
+            messageCellType == .oversizeTextMessage else {
+                return nil
+        }
+        guard let displayableText = viewItem.displayableText() else {
+                return nil
+        }
+        let messageBody = displayableText.fullText
+        guard messageBody.characters.count > 0  else {
+            return nil
+        }
+        return messageBody
+    }
+
     private func contentRows() -> [UIView] {
         var rows = [UIView]()
 
-        if message.attachmentIds.count > 0 {
+        if let messageBody = displayableTextIfText() {
+
+            self.messageBody = messageBody
+
+            let isIncoming = self.message as? TSIncomingMessage != nil
+
+            // UITextView can't render extremely long text due to constraints
+            // on the size of its backing buffer, especially when we're 
+            // embedding it "full-size' within a UIScrollView as we do in this view.
+            //
+            // TODO: We could use CoreText instead, or we could dynamically
+            //       manipulate the size/position of our UITextView to
+            //       reflect scroll state.
```

Comment:
This is a serious issue.  I'm going to make a card.  I'm not sure of its priority.